File: docs/examples/js/src/data_replication_mode.mjs
import { Client, ReplicationMode } from "reduct-js";

// Update replication mode without changing other settings
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });

// Pause to queue transactions without sending them
await client.setReplicationMode("my_replication", ReplicationMode.PAUSED);

// Disable to ignore new transactions
await client.setReplicationMode("my_replication", ReplicationMode.DISABLED);

// Re-enable to resume sending queued data
await client.setReplicationMode("my_replication", ReplicationMode.ENABLED);

File: versioned_docs/version-1.21.x/examples/rs/examples/data_replication_mode.rs
use reduct_rs::{ReductClient, ReductError, ReplicationMode};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Update replication mode without changing other settings
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Pause to queue transactions without sending them
    client
        .set_replication_mode("my_replication", ReplicationMode::Paused)
        .await?;

    // Disable to ignore new transactions
    client
        .set_replication_mode("my_replication", ReplicationMode::Disabled)
        .await?;

    // Re-enable to resume sending queued data
    client
        .set_replication_mode("my_replication", ReplicationMode::Enabled)
        .await?;

    Ok(())
}
